### PR TITLE
use new user input object for add user mutation

### PR DESCRIPTION
**Pull request by BobanL**
# FRONTEND PULL REQUEST

## Related Issue

- Part of #5945 - send permissions and facility set to addUser 

## Changes Proposed

- Use the new user input object to send to backend

## Additional Information

- Search results for users may fail for new users, so query can be used for fallback. Changes have been made to make this permanent.

## Testing

- Add a new user and make sure they are created as expected
- [Deployed in dev](https://dev.simplereport.gov/)

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java [hunk 1]
```diff
@@ -651,4 +630,15 @@ private void throwErrorIfEmpty(Stream<?> stream, String errorMessage) {
       throw new IllegalGraphqlArgumentException(errorMessage);
     }
   }
+
+  private User getUserOrThrowError(String username, String errorMessage) {
+    var searchUsersStream =
+        _client.listUsers(null, null, generateLoginSearchTerm(username), null, null).stream();
+    var user = searchUsersStream.findFirst();
+    if (user.isEmpty()) {
+      var qUsersStream = _client.listUsers(username, null, null, null, null).stream();
```
Comment (fzhao99) [inline backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java:639]: is there a convention / meaning for prefixing the variable here with `q`? Asking for curiosity!

**Review by fzhao99 — APPROVED** [review]
lgtm!

File: backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java (hunk 1 above)
Comment (BobanL) [reply backend/src/main/java/gov/cdc/usds/simplereport/idp/repository/LiveOktaRepository.java:639]: That's just the name of the parameter in [okta's documentation](https://developer.okta.com/docs/reference/api/users/#content-type-header-fields-2:~:text=FALSE-,q,FALSE,-limit).

**Review by emyl3 — APPROVED** [review]
tested on dev -- LGTM!

**Pull request merged**